### DISCUSSION TITLE
HANA & Netweaver pre-configuration roles for SLES

### DISCUSSION THREAD

**Discussion by mr-stringer** [sap-linuxlab/community.sap_install#221] [Ideas]
The HANA and Netweaver configuration roles work by process looping through a number of task files which use Ansible to implement SAP notes.

The preferred method of HANA and Netweaver pre-configuration for SLES to install saptune and implement the desired saptune profile.  We at SUSE already have some internal Ansible playbooks that could server as a basis for this.  

Would the maintainer entertain a PR that alters the SLES tasks section of the above roles so that saptune is used.  I think saptune should be optional, but perhaps the default SLES option.  If saptune is not used, then notes could be implemented as they are now.

I'd be happy to write the code.  I'm just looking for consensus before I work on a PR.

**Comment by berndfinger** [comment]
Hi @mr-stringer,

I checked and found that currently, no SAP notes related tasks are provided and pulled in during the configuration phase for SLES in role `sap_general_preconfigure`. But one note is mentioned in role `sap_netweaver_preconfigure` and several notes are mentioned in role `sap_hana_preconfigure`. I suggest to not just use the note numbers but also their versions, as in variable  `__sap_hana_preconfigure_sapnotes_versions_ppc64le`, for example.

I don't see any problems if the configuration steps for SLES only implement saptune and not use separate notes related yml files. If you decide to implement the separate notes related yml files as well, we could add a variable called ` sap_XXX_preconfigure_use_saptune_for_sles` (as an example; replace "XXX" by "general", "netweaver" or "hana" as appropriate), defaulting to `yes`, which one could use to call the SAP notes code and not use saptune.

Please also contact @rainerleber and discuss your proposal with him, as he had provided the initial code changes of the preconfigure roles for SLES.

All preconfigure roles support an assertion run, which does not change anything but just verifies that a managed node is compliant with all required installation and configuration steps. This is not the same as the Ansible check mode but you might call it "extended check mode".

I leave it to you to decide if you want to implement such a functionality for SLES as well but I strongly vote for it so that the roles behave similarly on both platforms. If you decide not to implement the assert functionality for SLES, please make sure that there is a warning message which explains that the assertions will not be done for SLES.

**Reply by mr-stringer** [reply]
I want to be as consistent to your standard as possible so I will implement an assert mode too.

**Comment by mr-stringer** [comment]
Hi @rainerleber, can you please review the current discussion and provide feedback to whether you are happy for saptune to be implemented in the role?

**Comment by rainerleber** [comment]
Hi @mr-stringer yes i am totally fine with saptune implementation and very happy to work with you on upcoming changes :-)

**Comment by mr-stringer** [comment]
Hello again.

I've had a crack at replacing the sapnote implementation in the SAP HANA preconfigure role so that SLES15 uses saptune instead.  The work in progress branch is [here](https://github.com/mr-stringer/community.sap_install/tree/hana-saptune).

See the files ./roles/sap_hana_preconfigure/tasks/SLES.  So far I've written files for installation and configuration.  If these look about right, I'll continue and do the assert ones.  

Any criticism, recommendations, etc, is most welcome.

Steve.
